test/DynamoCoreTests/CoreTests.cs

```diff
@@ -51,6 +51,26 @@ public void CanAddANodeByName()
 
         [Test]
         [Category("UnitTests")]
+        public void WorkspaceModelHasCorrectDependencies()
+        {   
+            var addNode = new DSFunction(CurrentDynamoModel.LibraryServices.GetFunctionDescriptor("+"));
+            var ws = this.CurrentDynamoModel.CustomNodeManager.CreateCustomNode("someNode", "someCategory", "");
+            var csid =  (ws as CustomNodeWorkspaceModel).CustomNodeId;
+            var customNode = this.CurrentDynamoModel.CustomNodeManager.CreateCustomNodeInstance(csid);
+
+            Assert.AreEqual(0, CurrentDynamoModel.CurrentWorkspace.Dependencies.ToList().Count());
+
+            CurrentDynamoModel.AddNodeToCurrentWorkspace(customNode,false);
+            CurrentDynamoModel.CurrentWorkspace.AddAndRegisterNode(addNode, false);
+            Assert.AreEqual(1, CurrentDynamoModel.CurrentWorkspace.Dependencies.ToList().Count());
+            //assert that we still only record one dep even though custom node is in graph twice.
+            CurrentDynamoModel.AddNodeToCurrentWorkspace(customNode, false);
+            Assert.AreEqual(1, CurrentDynamoModel.CurrentWorkspace.Dependencies.ToList().Count());
```

Comment:
Can we check the node guid in hashset is same as the original `customNode` id as well?